iotc-c-generic-sdk/azure-sdk-impl/src/azure_sdk_client.c

```diff
@@ -55,14 +81,21 @@ static char *file_to_string(const char *filename) {
 static void client_deinit() {
     is_client_active = false;
     if (device_ll_handle) {
+        IoTHubDeviceClient_LL_DoWork(device_ll_handle);
+        ThreadAPI_Sleep(10);
         IoTHubDeviceClient_LL_Destroy(device_ll_handle);
         device_ll_handle = NULL;
     }
+    free(dps_data.device_id);
+    free(dps_data.iothub_uri);
+    memset(&dps_data, 0, sizeof(dps_data));
+    prov_dev_security_deinit();
     c2d_msg_cb = NULL;
     status_cb = NULL;
 }
-static void send_confirm_callback(IOTHUB_CLIENT_CONFIRMATION_RESULT result, void* userContextCallback) {
-    (void)userContextCallback;
+
+static void send_confirm_callback(IOTHUB_CLIENT_CONFIRMATION_RESULT result, void *userContextCallback) {
```

Comment:
Rename *userContextCallback* to *user_context_callback*.

Reply:
Right... Carried this over from Azure's sample code. Thanks.